### ISSUE TITLE
Extending Packer plugin development examples

### ISSUE THREAD

**Issue by nywilken** [hashicorp/packer#8952]
The existing Extending Packer documentation provides context around building custom plugins such as a new Builder, Provisioner, or PostProcesser. With the focus on providing enough details to get a developer started. But it does easily point to how those constructs are used in an existing plugin. 

It would be great to expand on the documentation with an example or two for how to contribute to an existing plugin. Possibly using a small and more complex plugin code base to explain how and when to add a new configuration property vs creating a new step to interact with some upstream API. Some other ideas that come to mind are as follows: 

- How data is shared between statebags.
- How a plugin is structured into separate packages (common, plugin specific).
- Brief overview of testing a plugin: unit test vs acceptance test

**Comment by azr** [comment]
That a good idea and I think the [Guide on making a plugin HCL2 enabled](https://packer.io/guides/hcl/component-object-spec) should be part of it too ! 🙂

**Comment by adeniyistephen** [comment]
@SwampDragons @nywilken Hi, is this issue still open, i would love to work on it, kindly provide input and guids

**Comment by SwampDragons** [comment]
I appreciate the enthusiasm but this may end up being better done by one of the core maintainers -- it's going to be a lot of tedious complex documentation, and it relates to a part of the code that is currently changing rapidly. 

If you're interested in contributing docs, I can probably point you to a few easier things to work through; otherwise if you want to contribute code, let me know what you preferred builders/provisioners are and I can try to point you to some reasonably beginner-friendly issues.

**Comment by adeniyistephen** [comment]
Thanks for the feed back. I would prefer to contribute to docs for now.

**Comment by adeniyistephen** [comment]
> I appreciate the enthusiasm but this may end up being better done by one of the core maintainers -- it's going to be a lot of tedious complex documentation, and it relates to a part of the code that is currently changing rapidly.
> 
> If you're interested in contributing docs, I can probably point you to a few easier things to work through; otherwise if you want to contribute code, let me know what you preferred builders/provisioners are and I can try to point you to some reasonably beginner-friendly issues.

Yes i would really love to work on the documentations. Kindly point me to them and provide inputs. Thanks :)
